pennylane/qchem/molecule.py

```diff
@@ -97,6 +97,16 @@ def __init__(
 
         self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols, load_data)
 
+        self.nuclear_charges = [atomic_numbers[s] for s in self.symbols]
+
+        self.n_electrons = sum(np.array(self.nuclear_charges)) - self.charge
+
+        if self.n_electrons % 2 == 1 or self.mult != 1:
+            raise ValueError(
```

Comment:
Do we really need to raise a `ValueError`, or a simple `warning` would be sufficient?

Reply:
It should be an error because for radicals, e.g., H3 radical, the computed HF energy will be wrong.